src/keyszer/xorg.py

```diff
@@ -32,6 +32,10 @@ def get_xorg_context():
         window = get_actual_window(input_focus)
         if window:
             wm_name = window.get_wm_name()
+            # Sometimes legacy WM_NAME attribute is encoded as COMPOUND_TEXT,
+            # causing empty byte object to return, instead of string. To fix:
+            if isinstance(wm_name, bytes) or len(wm_name) == 0:
+                wm_name = window.get_full_text_property(343)    # use _NET_WM_NAME string instead
```

Comment:
343 is a magic number... can we not ask the server for the property reference and then hold onto that like I've seen done elsewhere?

Also, if WM_NAME is legacy, should the correct behavior perhaps be just trying _NET_WM_NMAE *first* and only falling back to the legacy attribute where there is no new attribute?

Reply:
The magic number comes back from `get_atom` (`intern_atom` appears to be an alias) when you give it "_NET_WM_NAME". Oddly, this is not listed in the defined "atoms" file, which seems to be why I was having such a hard time just calling it up in the same way as WM_NAME. 

I haven't found a more direct way to get it by name, but I could just embed the thing (I think it was something like "_display.get_atom("_NET_WM_NAME") ) that returns "343" inside the `get_full_text_property`. It would be the same result, just more explicit. 

Or I can keep looking for a more direct way to call it. 

> Also, if WM_NAME is legacy, should the correct behavior perhaps be just trying _NET_WM_NMAE first and only falling back to the legacy attribute where there is no new attribute?

I would be pretty fine with that. The alternate attribute should always just show the same thing in proper string form, if I understand the reason for its creation correctly. But most likely it would never have any reason to fall back to WM_NAME. Even if it ever did, I'd still need to include the check for it being a bytes object and attempt to decode it to a string in the case that it isn't empty. Since that seems incredibly unlikely the fall-back would just be to return an empty string instead of an empty bytes object. 

It feels like both the easiest and safest fix is just to completely ignore the existence of the "legacy" WM_NAME and return only _NET_WM_NAME in its place without any quibbling. 

I'll change it to this. Works for me. 

```py
            wm_name = window.get_full_text_property(_display.get_atom("_NET_WM_NAME"))
```